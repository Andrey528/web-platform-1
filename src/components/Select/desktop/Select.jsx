import React, { useState } from 'react';
import block from 'bem-cn';
import PropTypes from 'prop-types';
import SVGInlive from 'react-svg-inline';

import ArrowSVG from './img/arrow.svg';
import './Select.scss';

const Select = ({ items, activeItem, onChange }) => {
  const b = block('select');
  const [isOpen, changeOpen] = useState(false);
  const itemsList = items.map(item => <li
    key={item.value}
    className={b('item')}
    onClick={() => onChange(item.value)}
  >
    {item.name}
  </li>);
  return (
    <div
      className={b({ open: isOpen })}
      onClick={() => changeOpen(!isOpen)}
      onMouseLeave={() => changeOpen(false)}
    >
      <div className={b('item', { current: true })}>
        {activeItem.name}
        <SVGInlive className={b('arrow').toString()} svg={ArrowSVG} />
      </div>
      {isOpen && <ul className={b('items-list')}>{itemsList}</ul>}
    </div>
  );
};

Select.propTypes = {
  items: PropTypes.array.isRequired,
  activeItem: PropTypes.object.isRequired,

  onChange: PropTypes.func.isRequired,
};

export default Select;