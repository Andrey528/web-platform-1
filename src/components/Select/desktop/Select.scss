@import 'shared/style/varDesktop.scss';

.select {
  width: 20rem;
  border-radius: 1.1rem;
  border: 1px solid $grey-border;
  box-sizing: border-box;
  background: #ffffff;
  position: relative;
  user-select: none;

  &_open {
    border-radius: 1.1rem 1.1rem 0 0;
    border-bottom: none;
  }
  
  &__item {
    width: 100%;
    height: 2.125rem;
    position: relative;
    display: flex;
    align-items: center;
    cursor: pointer;
    padding: 0 1.0938rem;
    box-sizing: border-box;

    &:hover {
      background: #eeeeee;
    }

    &_current:hover {
      background: none;
    }
  }

  &__arrow {
    position: absolute;
    display: block;
    width: 0.5rem;
    top: 0.6875rem;
    right: 1rem;
    transition: transform .3s ease;
  }

  &_open &__arrow {
    transform: rotate(180deg);
  }

  &__items-list {
    position: absolute;
    top: 2.125rem;
    left: -1px;
    border: 1px solid $grey-border;
    border-top: none;
    border-radius: 0 0 1.1rem 1.1rem;
    background: #ffffff;
    width: 20rem;
    box-sizing: border-box;
    z-index: 2;
    max-height: 21rem;
    overflow: auto;

    &::-webkit-scrollbar {
      width: 3px;
      height: 90%;
      scroll-margin-right: 0.5rem;
    }
    &::-webkit-scrollbar-thumb {
      width: 3px;
      background: $grey-border;
      cursor: pointer;
    }
  }
}