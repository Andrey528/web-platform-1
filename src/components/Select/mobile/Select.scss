@import 'shared/style/varMobile.scss';

.select {
  width: 100%;
  border-bottom: 3px solid $select-border;
  box-sizing: border-box;
  background: $select-main-background;
  position: relative;
  user-select: none;
  color: white;

  &_open {
    border-radius: 1.1rem 1.1rem 0 0;
    border-bottom: none;
  }
  
  &__item {
    width: 100%;
    height: 2.125rem;
    position: relative;
    display: flex;
    align-items: center;
    cursor: pointer;
    padding: 0 0.60938rem;
    box-sizing: border-box;

    &:hover {
      background: #eeeeee;
    }

    &_current:hover {
      background: none;
    }
  }

  &__arrow {
    position: absolute;
    display: flex;
    align-items: center;
    justify-content: center;
    width: 0.5rem;
    right: 1rem;
    transition: transform .3s ease;
  }

  &__arrow-svg {
    height: 0.571rem;
    width: 0.857rem;

  }

  &_open &__arrow {
    transform: scaleY(-1);
  }

  &__items-list {
    position: absolute;
    top: 2.125rem;
    left: -1px;
    border: 1px solid $select-border;
    border-top: none;
    border-radius: 0 0 1.1rem 1.1rem;
    background: $select-main-background;
    width: 100%;
    z-index: 2;
    max-height: 21rem;
    overflow: auto;

    &::-webkit-scrollbar {
      width: 3px;
      height: 90%;
      scroll-margin-right: 0.5rem;
    }
    &::-webkit-scrollbar-thumb {
      width: 3px;
      background: #C4C4C4;
      cursor: pointer;
    }
  }
}