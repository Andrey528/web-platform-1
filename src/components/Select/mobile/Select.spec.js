import React from 'react';
import { shallow } from 'enzyme';
import block from 'bem-cn';

import Select from './Select';

describe('Select tests', () => {
  const b = block('select');
  const props = {
    items: [{ name: 'testName', value: 'testValue' }],
    activeItem: { name: 'test' },
    onChange: jest.fn(),
  };

  describe('Select with value', () => {
    const select = shallow(<Select {...props} />);

    it('render properly', () => {
      expect(select).toMatchSnapshot();
    });

    it('check render empty ul', () => {
      expect(select.find('ul')).toEqual({});
    });
  });

  describe('Select have modificators', () => {
    it('Div 0 have different modificators', () => {
      const className = b({ open: false });
      const select = shallow(<Select {...props} />);

      expect(select.find('div').at(0).props().className.toString()).toEqual(className.toString());
    });

    it('Div 1 have different modificators', () => {
      const className = b('item', { current: true });
      const select = shallow(<Select {...props} />);

      expect(select.find('div').at(1).props().className.toString()).toEqual(className.toString());
    });
  });
});