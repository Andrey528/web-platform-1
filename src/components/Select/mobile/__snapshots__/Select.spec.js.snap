// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select tests Select with value render properly 1`] = `
<div
  className={
    Object {
      "has": [Function],
      "is": [Function],
      "mix": [Function],
      "split": [Function],
      "state": [Function],
      "toString": [Function],
    }
  }
  onClick={[Function]}
  onMouseLeave={[Function]}
>
  <div
    className={
      Object {
        "has": [Function],
        "is": [Function],
        "mix": [Function],
        "split": [Function],
        "state": [Function],
        "toString": [Function],
      }
    }
  >
    test
    <SVGInline
      className="select__arrow"
      classSuffix="-svg"
      cleanup={Array []}
      cleanupExceptions={Array []}
      component="span"
      svg="test-file-stub"
    />
  </div>
</div>
`;
